python/spark_sklearn/keyed_models.py

```diff
@@ -328,7 +398,10 @@ def fitEstimator(_, pandasDF):
             # Potential optimization - del pandasDF
 
             estimatorClone = sklearn.base.clone(estimator)
-            estimatorClone.fit(X, y)
+            if y is None:
```

Comment:
You are right but not for the right reasons. This will work as you would expect:
```
1 if np.array([]) else 0
1 if np.array([1.0]) else 0
```
but this will fail at runtime with a message about truth value:
```
1 if np.array([1.0, 0.0]) else 0
```
The reason being that the evaluation of truth of an array returns an array of booleans, not a boolean.